### PR TITLE
fee burner wrapper with fee sharing

### PR DESCRIPTION
**Pull request by ilanDoron**

### CODE REVIEW

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    //add reserve pending data
+    struct AddReserveData {
+        address reserve;
+        uint    feeBps;
+        address kncWallet;
+    }
+
+    AddReserveData private addReserve;
+
+    //wallet fee parameters
```
Comment (talbaneth) [inline contracts/wrapperContracts/WrapFeeBurner.sol]: consider writing "wallet fee **pending** parameters", to align with other comments.
This will be clearer since you don't used the "pending" prefix in the variable names.

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        require(kncWallet != address(0));
+        require(feeBps > 0);
+
+        addReserve.reserve = _reserve;
+        addReserve.feeBps = feeBps;
+        addReserve.kncWallet = kncWallet;
+        setNewData(ADD_RESERVE_INDEX);
+    }
+    
+    function approveAddReserveData(uint nonce) public onlyOperator {
```
Comment (talbaneth) [inline contracts/wrapperContracts/WrapFeeBurner.sol]: also here - consider moving approveAddReserveData to after getPendingAddReserveData and getAddReserveSignatures.

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+    function getAddReserveSignatures() public view returns (address[] signatures) {
+        uint nonce;
+        (signatures, nonce) = getDataTrackingParameters(ADD_RESERVE_INDEX);
+        return(signatures);
+    }
+
+    //wallet fee
+    /////////////
+    function setPendingWalletFee(address wallet, uint feeInBps) public onlyOperator {
```
Comment (talbaneth) [inline contracts/wrapperContracts/WrapFeeBurner.sol]: consider renaming feeInBps  to feeBps, didn't see the "In" style used elsewhere in this contract.

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        require(wallet != address(0));
+        require(feeInBps > 0);
+        walletFee.walletAddress = wallet;
```
Comment (talbaneth) [inline contracts/wrapperContracts/WrapFeeBurner.sol]: also require(feeInBps<10000)?

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+        walletFee.feeBps = feeInBps;
+        setNewData(WALLET_FEE_INDEX);
+    }
+
+    function approveWalletFeeData(uint nonce) public onlyOperator {
```
Comment (talbaneth) [inline contracts/wrapperContracts/WrapFeeBurner.sol]: same comment as above regarding location of approveWalletFeeData()

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+    function getWalletFeeSignatures() public view returns (address[] signatures) {
+        uint nonce;
+        (signatures, nonce) = getDataTrackingParameters(WALLET_FEE_INDEX);
+        return(signatures);
+    }
+
+    //tax parameters
+    ////////////////
+    function setPendingTaxParameters(address _taxWallet, uint _taxFeeBps) public onlyOperator {
```
Comment (talbaneth) [inline contracts/wrapperContracts/WrapFeeBurner.sol]: underscore prefixes for parameters are not used in other places.

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+    //knc rate range
+    struct KncPerEth {
+        uint minRate;
+        uint maxRate;
+        uint pendingMinRate;
+        uint pendingMaxRate;
+    }
+
+    KncPerEth private kncPerEth;
```
Comment (talbaneth) [inline contracts/wrapperContracts/WrapFeeBurner.sol]: why is kncPerEth is private while in other places (like feeSharingWallets) we use internal?
I understand that private does not let you access the variable from other contracts, even not ones derived from this contract. o is it private intentionally?

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+    function setPendingTaxParameters(address _taxWallet, uint _taxFeeBps) public onlyOperator {
+        require(_taxWallet != address(0));
+        require(_taxFeeBps > 0);
+
+        taxData.wallet = _taxWallet;
+        taxData.feeBps = _taxFeeBps;
+        setNewData(TAX_DATA_INDEX);
+    }
+
+    function approveTaxData(uint nonce) public onlyOperator {
```
Comment (talbaneth) [inline contracts/wrapperContracts/WrapFeeBurner.sol]: maybe change location of approve, same as in other places.

File: contracts/wrapperContracts/WrapFeeBurnerWrapper.sol [hunk 9]
```diff
@@ -0,0 +1,28 @@
+pragma solidity ^0.4.18;
+
+
+import "../Withdrawable.sol";
+
+
+interface WrapFeeBurner{
```
Comment (talbaneth) [inline contracts/wrapperContracts/WrapFeeBurnerWrapper.sol]: for my understanding - what is the purpose of declaring this interface? (and interfaces in general)

File: web3deployment/basicDeployer.js [hunk 10]
```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env node
+
```
Comment (talbaneth) [inline web3deployment/basicDeployer.js]: what is the purpose of this script? how is it different than deployment.js?

File: web3deployment/basicDeployer.js [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+    .boolean('printPrivateKey')
+    .boolean('dontSendTx')
+    .argv;
+const web3 = new Web3(new Web3.providers.HttpProvider(rpcUrl));
+const solc = require('solc')
+
+const rand = web3.utils.randomHex(7);
+
+//const privateKey = web3.utils.sha3("in joy we trust" + rand);
+const privateKey = '0x1fdeefabe80cb822147afa9eadb9c29966788f7a5562dacbef2499175aca25a7';
```
Comment (talbaneth) [inline web3deployment/basicDeployer.js]: is it ok to put the private key like this?

File: web3deployment/mainnet_staging.json [hunk 12]
```diff
@@ -442,12 +442,20 @@
       "alerter": []
     },
     "FeeBurner": {
-      "admin": "0xd0643BC0D0C879F175556509dbcEe9373379D5C3",
+      "admin": "0xBe401c3cf8528DB1B963e2E40827a2E0e1d98Ee4",
```
Comment (talbaneth) [inline web3deployment/mainnet_staging.json]: is this the new wrapper contract address?

File: web3deployment/mainnet_staging.json [hunk 13]
```diff
@@ -442,12 +442,20 @@
...
       "operator": [
         "0xd0643BC0D0C879F175556509dbcEe9373379D5C3"
       ],
       "alerter": []
     },
+    "FeeBurner Wrapper": {
+      "admin": "0x7C8cfF2c659A3eE23869497a56129F3da92E8F38",
```
Comment (talbaneth) [inline web3deployment/mainnet_staging.json]: is this Yaron?

File: contracts/wrapperContracts/WrapFeeBurner.sol (hunk 7 above)
Comment (ilanDoron) [reply contracts/wrapperContracts/WrapFeeBurner.sol]: well.
Actually it has same meaning hear. since we don't intend to inherit this contract anyway.
So I assume it better be the same.
But not sure its wort
h a new deploy

File: web3deployment/mainnet_staging.json (hunk 13 above)
Comment (ilanDoron) [reply web3deployment/mainnet_staging.json]: yes

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 14]
```diff
@@ -0,0 +1,227 @@
...
+        (signatures, nonce) = getDataTrackingParameters(KNC_RATE_RANGE_INDEX);
+        return(signatures);
+    }
+
+    //set reserve data
+    //////////////////
+    function setPendingReserveData(address _reserve, uint feeBps, address kncWallet) public onlyOperator {
+        require(_reserve != address(0));
+        require(kncWallet != address(0));
+        require(feeBps > 0);
```
[the remark this answers is not in the document]
Comment (ilanDoron) [reply contracts/wrapperContracts/WrapFeeBurner.sol]: done

File: contracts/wrapperContracts/WrapperBase.sol [hunk 15]
```diff
@@ -1,17 +1,15 @@
...
-import "../PermissionGroups.sol";
 
 
 contract WrapperBase is Withdrawable {
 
     PermissionGroups public wrappedContract;
 
     struct DataTracker {
-        address [] approveSignatureArray;
+        address[] approveSignatureArray;
```
Comment (yaronvel) [inline contracts/wrapperContracts/WrapperBase.sol]: revert the change please

File: contracts/wrapperContracts/WrapFeeBurner.sol [hunk 16]
```diff
@@ -134,20 +133,20 @@ contract WrapFeeBurner is WrapperBase {
 
     //set reserve data
     //////////////////
-    function setPendingReserveData(address _reserve, uint feeBps, address kncWallet) public onlyOperator {
-        require(_reserve != address(0));
+    function setPendingReserveData(address reserveRes, uint feeBps, address kncWallet) public onlyOperator {
```
Comment (yaronvel) [inline contracts/wrapperContracts/WrapFeeBurner.sol]: why reserveRes
